src/wallet/rpcwallet.cpp

```diff
@@ -2645,10 +2646,33 @@ UniValue listwallets(const JSONRPCRequest& request)
 
         LOCK(pwallet->cs_wallet);
 
-        obj.push_back(pwallet->GetName());
+        loaded.push_back(pwallet->GetName());
+        loaded_set.insert(pwallet->GetName());
     }
 
-    return obj;
+    UniValue available(UniValue::VARR);
+    fs::path walletdir = GetDataDir();
+    const unsigned char bdb_magic[4] = {0x62, 0x31, 0x05, 0x00}; // Berkeley DB Btree magic bytes
+
+    for (fs::directory_iterator it(walletdir); it != fs::directory_iterator(); ++it) {
+        std::ifstream file(it->path().string(), std::ios::binary);
+        file.seekg(12, std::ios::beg); // Magic bytes start at offset 12
+        unsigned char fileBytes[4] = {0};
+
+        file.read((char*)fileBytes, sizeof(fileBytes)); // Read 4 bytes of file to compare against magic
+
+        if (fs::is_regular_file(*it) && !fs::is_symlink(*it) && file.is_open() && memcmp(fileBytes, bdb_magic, sizeof(bdb_magic)) == 0) {
```

Comment:
Some of these conditions can be moved before seekg (just `continue`. Only the compare is needed.

Could use 64 bit int instead?

Reply:
Fixed the conditional, but I think the char array is clearer and perhaps safer than using an int